sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -66,19 +66,25 @@ object ScalaReflection extends ScalaReflection {
    */
   def dataTypeFor[T : TypeTag]: DataType = dataTypeFor(localTypeOf[T])
 
+  private[catalyst] def isSubtype(tpe1: `Type`, tpe2: `Type`): Boolean = {
```

Comment:
can we add some comments to explain it and link to the scala bug report?

Reply:
👍 Added a comment